packages/stripe-integration/src/stripe-upe/stripe-ocs-payment-strategy.ts

```diff
@@ -0,0 +1,413 @@
+} from '@bigcommerce/checkout-sdk/payment-integration-api';
+
+import formatLocale from './format-locale';
+import { isStripePaymentEvent } from './is-stripe-payment-event';
+import { isStripeUPEPaymentMethodLike } from './is-stripe-upe-payment-method-like';
+import {
+    StripeElement,
+    StripeElements,
+    StripeElementType,
+    StripeEventType,
+    StripeStringConstants,
+    StripeUPEAppearanceOptions,
+    StripeUPEClient,
+} from './stripe-upe';
+import StripeUPEPaymentInitializeOptions, {
+    WithStripeUPEPaymentInitializeOptions,
+} from './stripe-upe-initialize-options';
+import StripeUPEIntegrationService from './stripe-upe-integration-service';
+import StripeUPEScriptLoader from './stripe-upe-script-loader';
+
+export default class StripeOCSPaymentStrategy implements PaymentStrategy {
+    private stripeUPEClient?: StripeUPEClient;
+    private stripeElements?: StripeElements;
+    private selectedMethodId?: string;
+
+    constructor(
+        private paymentIntegrationService: PaymentIntegrationService,
+        private scriptLoader: StripeUPEScriptLoader,
+        private stripeUPEIntegrationService: StripeUPEIntegrationService,
+    ) {}
+
+    async initialize(
+        options: PaymentInitializeOptions & WithStripeUPEPaymentInitializeOptions,
+    ): Promise<void> {
+        const { stripeupe, methodId, gatewayId } = options;
+
+        if (!stripeupe?.containerId) {
+            throw new NotInitializedError(NotInitializedErrorType.PaymentNotInitialized);
+        }
+
+        if (!gatewayId) {
+            throw new InvalidArgumentError(
+                'Unable to initialize payment because "gatewayId" argument is not provided.',
+            );
+        }
+
+        await this._initializeStripeElement(stripeupe, gatewayId, methodId).catch((error) =>
+            stripeupe.onError?.(error),
+        );
+
+        this.stripeUPEIntegrationService.initCheckoutEventsSubscription(
+            gatewayId,
+            methodId,
+            stripeupe,
+            this.stripeElements,
+        );
+    }
+
+    async execute(orderRequest: OrderRequestBody, options?: PaymentRequestOptions): Promise<void> {
+        const { payment, ...order } = orderRequest;
+        const { methodId, gatewayId } = payment || {};
+
+        if (!this.stripeUPEClient) {
+            throw new NotInitializedError(NotInitializedErrorType.PaymentNotInitialized);
+        }
+
+        if (!gatewayId || !methodId) {
+            throw new InvalidArgumentError(
+                'Unable to initialize payment because "gatewayId" or "methodId" argument is not provided.',
+            );
+        }
+
+        const state = this.paymentIntegrationService.getState();
+        const { isStoreCreditApplied } = state.getCheckoutOrThrow();
+
+        if (isStoreCreditApplied) {
+            await this.paymentIntegrationService.applyStoreCredit(isStoreCreditApplied);
+        }
+
+        await this.paymentIntegrationService.loadPaymentMethod(gatewayId, {
```

Comment:
Should payment method be loaded each time when shopper clicks on stripe button? Can we check it the method was loaded previously (paymentIntegrationService.getPaymentMethod(...)) and if not then load it here?

Reply:
As I mentioned before, in this case `loadPaymentMethod` does not only load method config, but also triggers Payment Intent updates on BE side. Thats why we need this request.

Reply:
Should we add a comment here just to avoid removing this line by mistake or during refactoring (by someone else) in the future?

Reply:
Add new method with a comment? to make this logic more transparent
https://github.com/bigcommerce/checkout-sdk-js/commit/bffde45e9422f860b94448c581bbad816d78f3fb